woof-code/rootfs-skeleton/bin/mount-pup

```diff
@@ -86,19 +86,124 @@ elif [ "$NTFS_FLAG" = "yes" ];then
 
   [ ! -d /tmp/ntfs-errors ] && mkdir -p /tmp/ntfs-errors
 
+  case "$@" in
   case "$@" in
     *"-o "*) #don't override -o
-      ntfs-3g "${@}" || busybox mount "${@}"
+    
+    #check for NTFS3 modules
+    NTFS3_FOUND="$(zcat /proc/config.gz 2>/dev/null | grep "CONFIG_NTFS3_FS=y")"
+      
+    if [ "$NTFS3_FOUND" != "" ]; then
+     USE_NTFS3="y"
+    elif [ -e /sys/modules/ntfs3 ]; then
+     USE_NTFS3="y"
+    elif [ "$(cat /proc/filesystems | grep ntfs3)" != "" ]; then
+     USE_NTFS3="y"
+    elif [ -e /sys/fs/ntfs3 ]; then
+     USE_NTFS3="y" 
+    elif [ -e /lib/modules/$(uname -r)/kernel/fs/ntfs3 ]; then
+     USE_NTFS3="y" 
+    else
+     USE_NTFS3=""
+    fi
+     
+     NTFS_3G="y" 
+      
+      if [ "$USE_NTFS3" != "" ]; then
+       #mount using ntfs3
+       busybox mount "$(echo "${@}" | sed -e 's#\ ntfs\ #\ ntfs3 \#')"
+       RETVAL=$?
+       	   if [ $RETVAL -eq 0 ]; then
+	    #mount success
+	    NTFS_3G="" 
+	   else
+	   
+	     for arg1 in ${@}
+	     do
+	      #Look for device or mountpoint
+	      if [ "$(echo "$arg1" | grep "/")" != "" ]; then
+	       ptn="$arg1"
+	       break
+	      fi
+	     done
+	     
+	     if [ "$ptn" != "" ]; then
+	      #mounted but with flaws
+	      [ "$(mount | grep -ow "$ptn")" != "" ] && NTFS_3G="" 
+	     fi
+	     
+	   fi
+      fi	
+      
+      if [ "$NTFS_3G" != "" ]; then
+       #fallback mount ntfs using ntfs-3g
+       ntfs-3g "${@}" || busybox mount "${@}"
+       RETVAL=$?
+      fi
+      
       RETVAL=$?
+      
       ;;
     *)
       #screen out all the options...
       CMDPRMS="`echo -n "$*" | tr '\t' ' ' | tr -s ' ' | tr ' ' '\n' | grep '^/' | tr '\n' ' '`"
       #kirk advised these options so Rox will not complain about file
       #permissions when copy a file to a ntfs partition...
-      [ -f /tmp/ntfs-errors/ntfsmnterr${MYPID}.txt ] && rm -f /tmp/ntfs-errors/ntfsmnterr${MYPID}.txt
+     
+     [ -f /tmp/ntfs-errors/ntfsmnterr-${MYPID}.txt ] && rm -f /tmp/ntfs-errors/ntfsmnterr-${MYPID}.txt
+      
+      NTFS3_FOUND="$(zcat /proc/config.gz 2>/dev/null | grep "CONFIG_NTFS3_FS=y")"
```

Comment:
You can just try ntfs3 and fall back to ntfs-3g. It will fail fast, because ntfs3 is an invalid parameter. This long detection sequence will slow down mounting.

Reply:
Not advisable. ntfs3 might threw exit code number but still mounting ntfs was sucessful. I just remove zcat process to make detection faster.